TKGame/Entity.cs

```diff
@@ -79,11 +78,17 @@ public Vector2 Size
                         {
                             // Player is above the wall
                             Position.Y -= (int)depth.Y;
+                            if(this is Player)
+                            {
+                                Player.Instance.CollidedVertically = true;
+                                Player.Instance.IsOnGround = true;
```

Comment:
Does Player.CollidedVertically check for hitting a ceiling or floor or both? This call tells me that they hit the ground but the bool CollidedVertically implies hitting the floor or the ceiling. 
It's in the right spot for the check so yes it will be when it hits the floor but the bool name could be a bit more explicit in that